websites/I/Instagram/presence.ts

```diff
@@ -1,37 +1,199 @@
+				presenceData.largeImageKey = await getShortURL(image.src);
+
+			presenceData.buttons = [
+				{
+					label: "View Post",
+					url: `https://www.instagram.com/${path[1]}/${path[2]}`
+				}
+			];
+		} else delete presenceData.buttons;
+	} else if (pathname.startsWith("/explore"))
 		presenceData.details = "Exploring...";
-	else if (document.location.pathname.startsWith("/nametag"))
+	else if (pathname.startsWith("/nametag"))
 		presenceData.details = "Viewing nametag";
 	else if (
-		document.location.pathname.startsWith("/direct/inbox") ||
-		document.location.pathname.startsWith("/direct/t")
+		pathname.startsWith("/direct/inbox") ||
+		pathname.startsWith("/direct/t")
 	)
 		presenceData.details = "Direct Messages";
-	else {
-		// TODO: Check if the page is really a profile
-		const [, pathName] = document.location.pathname.split("/");
-		presenceData.details = "Viewing a profile";
-		presenceData.state = pathName;
+	else if (profileName?.textContent === path[1]) {
+		const profilePicture =
+			document.querySelector<HTMLImageElement>("img._6q-tv");
+
+		presenceData.details = `Viewing a Profile${settings.privacy ? "" : ":"}`;
+
+		if (!settings.privacy) {
+			presenceData.state = `${
+				document.querySelector("div.QGPIr h1")?.textContent ?? "Unknown"
+			} (@${profileName.textContent})`;
+
+			if (profilePicture)
+				presenceData.smallImageKey = await getShortURL(profilePicture.src);
+
+			presenceData.buttons = [
+				{
+					label: "View Profile",
+					url: `https://www.instagram.com/${path[1]}`
+				}
+			];
+		} else {
+			delete presenceData.state;
+			delete presenceData.smallImageKey;
+			delete presenceData.buttons;
+		}
 	}
+
+	if (document.querySelector("div.QY4Ed.P0xOK input.focus-visible")) {
+		presenceData.details = settings.privacy ? "Searching" : "Searching:";
+
+		if (!settings.privacy) {
+			presenceData.state = document.querySelector<HTMLInputElement>(
+				"div.QY4Ed.P0xOK input.focus-visible"
+			)?.value;
+		} else delete presenceData.state;
+	}
+
 	presence.setActivity(presenceData);
 });
+
+function getDateString(date: Date) {
+	const seconds = Math.abs(Date.now() - date.getTime()) / 1000;
+
+	switch (true) {
+		case seconds < 60:
+			return `${Math.floor(seconds)} second${seconds === 1 ? "" : "s"} ago`;
+		case seconds < 3600:
+			return `${Math.floor(seconds / 60)} minute${
+				seconds === 60 ? "" : "s"
+			} ago`;
+		case seconds < 86400:
+			return `${Math.floor(seconds / 3600)} hour${
+				seconds === 3600 ? "" : "s"
+			} ago`;
+		case seconds < 604800:
+			return `${Math.floor(seconds / 86400)} day${
+				seconds === 86400 ? "" : "s"
+			} ago`;
+		case seconds < 2419200:
+			return `${Math.floor(seconds / 604800)} week${
+				seconds === 604800 ? "" : "s"
+			} ago`;
+		case seconds < 29030400:
+			return `${Math.floor(seconds / 2419200)} month${
+				seconds === 2419200 ? "" : "s"
+			} ago`;
+		default:
+			return `${Math.floor(seconds / 29030400)} year${
+				seconds === 29030400 ? "" : "s"
```

Comment:
I’m sure this isn’t right, it’s very unlikely for the variable to have this specific value